### DISCUSSION TITLE
Convert pdf with stencils to best fitting jpeg as 1 jpeg for 1 page

### DISCUSSION THREAD

**Discussion by artofit** [ImageMagick/ImageMagick#3987] [Digital Image Processing]
I ran pdfimages -list in.pdf on a problematic pdf, here's the output for the first 2 pages:

`page   num  type   width height color comp bpc  enc interp  object ID x-ppi y-ppi size ratio
--------------------------------------------------------------------------------------------
   1     0 image    2352  1655  gray    1   8  jpeg   no        82  0   200   200 7537B 0.2%
   1     1 stencil  3536  2357  -       1   1  jbig2  no        83  0   400   400 51.6K 5.1%
   1     2 stencil   481   478  -       1   1  jbig2  no        84  0   400   400 3716B  13%
   1     3 stencil    65   534  -       1   1  jbig2  no        85  0   400   400  482B  11%
   1     4 stencil   117   479  -       1   1  jbig2  no        86  0   400   400  408B 5.8%
   2     5 image    2352  1655  gray    1   8  jpeg   no         5  0   200   200 4685B 0.1%
   2     6 stencil  3043  2999  -       1   1  jbig2  no         6  0   400   400 45.3K 4.1%
   2     7 stencil   674  2232  -       1   1  jbig2  no         7  0   400   400 8967B 4.8%
   3     8 image    2352  1655  gray    1   8  jpeg   no        12  0   200   200 15.8K 0.4%`

If I run
`pdftoppm -jpeg "in.pdf" 2ppm_150DPI_defaultsq75
`I do get one jpeg per page, in my case 8pages as 8 jpegs,
contrary to 
`pdfimages -j "in.pdf" pdfImg
`which will give 1 jpeg per image or stencil, which leads to 48 images (jpeg, jbig2) for 8 pages.

But how can I achieve the best automated jpeg output as 1 jpeg for 1page?
As the background image is 2352 x 1655, while stencil size varies,
I can run:
`pdftoppm -jpeg -scale-to-x 2352 -scale-to-y 1655 "in.pdf" scaled`

But this requires me to first know how much to scale.
How to automate that? Different scanners, different scanning params.

Any idea?

**Comment by snibgo** [comment]
This is forum for ImageMagick. Your question is not about ImageMagick, but about the Poppler programs "pdftoppm" and "pdfimages".

Do you have any questions about ImageMagick?

**Reply by artofit** [reply]
Yes, how can IM do this job? And how? :)
I rather puzzled by the automation.

I see the stencils as layers onto a underlying image.
So I think that I ought to scale to the underlying image.
But maybe it's a bad view & idea.

**Comment by snibgo** [comment]
> Yes, how can IM do this job?

What job? Please say what output you want.

Please post a link to your PDF.
